src/app/hooks/useSignBatchPsbtTx.ts

```diff
@@ -0,0 +1,81 @@
+import { ExternalSatsMethods, MESSAGE_SOURCE } from '@common/types/message-types';
+import useWalletSelector from '@hooks/useWalletSelector';
+import {
+  InputToSign,
+  psbtBase64ToHex,
+  signPsbt,
+} from '@secretkeylabs/xverse-core/transactions/psbt';
+import { decodeToken } from 'jsontokens';
+import { useLocation } from 'react-router-dom';
+import { SignMultipleTransactionOptions } from 'sats-connect';
+import useBtcClient from './useBtcClient';
+import useSeedVault from './useSeedVault';
+
+const useSignBatchPsbtTx = () => {
+  const { accountsList, network } = useWalletSelector();
+  const { search } = useLocation();
+  const { getSeed } = useSeedVault();
+  const params = new URLSearchParams(search);
+  const requestToken = params.get('signBatchPsbtRequest') ?? '';
+  const request = decodeToken(requestToken) as any as SignMultipleTransactionOptions;
+  const tabId = params.get('tabId') ?? '0';
+  const btcClient = useBtcClient();
+
+  const confirmSignPsbt = async (psbt) => {
```

Comment:
I just tried to unite those hooks into one, but since they use different event and sats method names there will be a lot of conditions and the code won't look that nice, are you sure that we should still do that?